pkg/template/template.go

```diff
@@ -69,10 +70,12 @@ type (
 		BldrInterfaceName       string
 		HndlrInterfaceName      string
 		Variety                 adptTmpl.TemplateVariety
-		ProcessReport           ProcessReportFn
-		ProcessCheck            ProcessCheckFn
-		ProcessQuota            ProcessQuotaFn
+		Evaluate                EvaluateFn
```

Comment:
@guptasu We should go back to integrated  ProcessXXX.

Eval function operates on one instance at a time. During report flow it is not convenient to tell this function to hydrate 5 instances; type casting issue again.

At present the *caller will NOT* respond any differently if the error is from the evaluator or a bit down stream. Operator will have to read the error text to respond correctly.

Generated repetitive code is fine.

I am now coding to the following interface. 

```
ProcessCheckFn func(ctx context.Context, instName string, instCfg proto.Message, attrs attribute.Bag, mapper expr.Evaluator, handler adapter.Handler) (adapter.CheckResult, error)

ProcessQuotaFn func(ctx context.Context, quotaName string, quotaCfg proto.Message, attrs attribute.Bag, mapper expr.Evaluator, handler adapter.Handler, args adapter.QuotaRequestArgs) (adapter.QuotaResult2, error)

ProcessReportFn func(instCfg map[string]proto.Message, attrs attribute.Bag, mapper expr.Evaluator, handler adapter.Handler) error

```